packages/compartment-mapper/test/test-main.js

```diff
@@ -22,8 +25,16 @@ const assertFixture = (t, { namespace, globals, globalLexicals }) => {
     typemodule,
     typemoduleImplied,
     typeparsers,
+    metaUrl,
   } = namespace;
 
+  if (testCategoryHint === 'Location') {
+    // matching any character where / or \ would be
+    t.regex(metaUrl, /fixtures-0.node_modules.app.main\.js$/);
+  } else {
+    t.is(metaUrl, null);
```

Comment:
The more I see this, the more `import.meta.url` feels like a misfeature and that we should nudge the Endo community in the direction of text and binary asset modules, which we already support.

I think it’s a desirable property of Endo that, if a program runs with `importLocation`, it’ll also run with `makeArchive` and `parseArchive`, and in the latter case, `import.meta.url` and `__dirname` are not useful. However, with asset modules, we can preserve that guarantee, since the necessary assets are explicitly retained.

That being said, I can see value in the feature for Node.js portability, and we in fact could really use this feature real-soon-now™ in Agoric’s codebase. So, may I recommend that we add an option to `importLocation` to enable `needsImportMetaUrl`?